compiler/ccgtypes.nim

```diff
@@ -971,19 +982,79 @@ proc isReloadable(m: BModule; prc: PSym): bool =
 proc isNonReloadable(m: BModule; prc: PSym): bool =
   return m.hcrOn and sfNonReloadable in prc.flags
 
+proc parseVFunctionDecl(val: string; name, params, retType: var string; isFnConst, isOverride: var bool) =
+  var afterParams: string
+  if scanf(val, "$*($*)$s$*", name, params, afterParams):
+    isFnConst = afterParams.find("const") > -1
+    isOverride = afterParams.find("override") > -1
+    discard scanf(afterParams, "->$s$* ", retType)
+
+proc parseExistingParams(params:string, parsedParams: var OrderedTable[string, string]) =
+  #instead of messing with all the logic in params we do a pass over them here to "happy" extract them
+  #Ideally genProcParams would be refactored to return a table. So this is a temporary solution.
+  let splitParams = params.multiReplace(("(", ""), (")", "")).split(", ")
```

Comment:
What?

Reply:
Trying to fix the scope once again. Should I refactor genProcParams to use an intermediate shared function that returns the table? Or should I create another specific function just for virtual?